frontend/src/pages/AdminDashboard/UpgradeUser/index.jsx

```diff
@@ -0,0 +1,67 @@
+import React, { useState, useEffect, useRef } from "react";
+import Input from "../../../components/shared/Input";
+import Button from "../../../components/shared/Button";
+import { useUpgradeUserForm } from "./validation";
+import { UpgradeUserAPI } from "./api";
+import "./style.css";
+
+const UpgradeUser = () => {
+  const [alert, setAlert] = useState(false);
+  const isFirstRender = useRef(true);
+  const {
+    values,
+    errors,
+    touched,
+    isSubmitting,
+    status,
+    handleChange,
+    handleBlur,
+    handleSubmit,
+  } = useUpgradeUserForm(UpgradeUserAPI);
+
+  useEffect(() => {
+    if (isFirstRender.current) {
+      isFirstRender.current = false; // Skip effect on the first render
+      return;
+    }
+    if (status.status == "success");
+    else {
+      setAlert(true);
+      setTimeout(() => {
+        setAlert(false);
+      }, 2000);
```

Comment:
the alert  message disappears automatically after 2 seconds. This might not provide enough time for the user to read the message especially if they are distracted or need more time to react.
suggestions:
either increase the duration to give users more time to see and understand the message or allow users to dismiss the alert manually.

Reply:
Thank you for pointing out these valid concerns regarding the alert behavior. I completely agree that the current implementation might not provide an ideal user experience, particularly in cases where the alert disappears too quickly or doesn't reappear for repeated errors.

I intentionally deferred resolving this issue for now because it relates to the overall project design and alert handling mechanism, not just this specific feature. I plan to address it comprehensively in a subsequent update where the alert system can be improved holistically for all parts of the application. make this simpler